CI: Specify token when checking out demo

ChinYikMing opened the pull request:
Similar [issue](https://github.com/ad-m/github-push-action/issues/170) is discussed. I am not very sure we can solve it by this method but let's give it a try.

See issue: #400

jserv reviewed (changes requested):
Improve the git commit message, addressing more references.

ChinYikMing committed:
CI: Specify token when checking out demo

The existing WebAssembly CI works on my repository, but it is failed
here. This is very strange but we still have to solve it.

Combining some dicussion in github-push-action's issue and my own
thought, when checking out demo, the persist-credentials is set to
false, thus the token used for further action should be specified.

See issue: #400

ChinYikMing commented:
> Improve the git commit message, addressing more references.

Done.
